http_server/request_testing/request_test.go

```diff
@@ -0,0 +1,112 @@
+package requesttesting
+
+import (
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"testing"
+)
+
+type handlerRecorder struct {
```

Comment:
This would normally be called recorderHandler (the more generic name to the right, as what you care about should be the first thing that you read).